crates/client/rpc/src/lib.rs

```diff
@@ -150,7 +158,39 @@ where
         u64::try_from(block.header().block_number).map_err(|e| format!("Failed to convert block number to u64: {e}"))
     }
 
-    /// Helper function to filter Starknet events provided a RPC event filter
+    /// Helper function to get Starknet block details
+    ///
+    /// # Arguments
+    ///
+    /// * `block_id` - The Starknet block id
+    ///
+    /// # Returns
+    ///
+    /// * `(block_events: Vec<EventWrapper>, block: Block)` - A tuple of the block events in
+    ///   block_id and an instance of Block
+    fn get_block_events(&self, block_id: u64) -> Result<(Vec<EventWrapper>, Block), StarknetRpcApiError> {
+        let substrate_block_hash =
+            self.substrate_block_hash_from_starknet_block(BlockId::Number(block_id)).map_err(|e| {
+                error!("'{e}'");
+                StarknetRpcApiError::BlockNotFound
+            })?;
+
+        let block = get_block_by_block_hash(self.client.as_ref(), substrate_block_hash).ok_or_else(|| {
+            error!("Failed to retrieve block");
+            StarknetRpcApiError::BlockNotFound
+        })?;
+        let block_events = self
+            .overrides
+            .for_block_hash(self.client.as_ref(), substrate_block_hash)
+            .events(substrate_block_hash)
+            .unwrap_or_else(|| {
+                info!("No events found in block {}", block_id);
```

Comment:
info! is probably too strong.
dbg! should be enough

Reply:
fixed